spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -1699,34 +1699,70 @@ struct ProductionVisitor
     }
 
     void operator()(const production::Skip& p) {
+        auto consumeFixedSize = [&](const Expression& size) {
+            assert(size.type().template isA<type::UnsignedInteger>());
+
+            auto n = builder()->addTmp("skip", size);
+            auto loop = builder()->addWhile(builder::greater(n, builder::integer(0U)));
+            pushBuilder(loop, [&]() {
+                pb->waitForInput(builder::integer(1U), "not enough bytes for skipping", p.location());
+                auto consume = builder()->addTmp("consume", builder::min(builder::size(state().cur), size));
+                pb->advanceInput(consume);
+                builder()->addAssign(n, builder::difference(n, consume));
+                builder()->addDebugMsg("spicy-verbose", "- skipped %u bytes (%u left to skip)", {consume, n});
+            });
+        };
```

Comment:
I added `ParserBuilder::skip` which just captures this body. I think you mentioned offline that this would allow further optimizations, but I seem unable to see them, could you point me what else would be needed? We e.g., cannot trim unconditionally since the `skip` field might be in a non-trimming context.